sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -74,7 +74,7 @@
 COMMON_VALID_METRICS = sorted(
     set.intersection(*map(set, neighbors.VALID_METRICS.values()))
 )  # type: ignore
-P = (1, 2, 3, 4, np.inf)
```

Comment:
`np.inf` must still be supported, here.

Reply:
Not sure how to solve it with the param validator, for the Minkowski metric param, the validator uses 
`"p": [Interval(Integral, 1, None, closed="left")]`
Integer cannot have open interval on `inf`. I guess it should be left as integer, it's also in the doc for the Nearest neighbor. Any ideas how to solve it?

Reply:
But it does not work for integers (Integral), as I mentioned above. If you set it to `closed="both"` you get this error:
`ValueError: right can't be None when closed == both for an interval over the integers.`
One way would be to set it to `Real` but I believe the Minkowski metric should be an integer. Any ideas?

Reply:
Uhm right, you need to make it `Real`.

Reply:
`p` can be a real I think (at least scipy accept floating number). For instance, it can be between 0 and 1:

https://en.wikipedia.org/wiki/Minkowski_distance

Reply:
Ok, I changed it to `Real`, also in the documentation of the `NearestNeighbor`.